Prepare APM agent configuration for production use

joshdover opened the pull request:
## Summary

- Allows APM agent to be used in distributable builds
- Removes version from serviceName and uses serviceVersion tag instead.
- Turns off HTTP header and body capture by default
- Adds filter function to remove username PII info (populated by ES tracing automatically)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77691/)
* Commit: 8c209cdaa61749d17a34ab8f38ac5db5c0f1802b

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/8c209cdaa61749d17a34ab8f38ac5db5c0f1802b) | diff | [baseline](https://github.com/elastic/kibana/commit/6f1b6907ba010b40a688f99def1c8fa497949008) |
| --- | --- | --- | --- |
| core | 134.6KB | +134.6KB | 0.0B |

#### distributable file count
| id | [value](https://github.com/elastic/kibana/commit/8c209cdaa61749d17a34ab8f38ac5db5c0f1802b) | diff | [baseline](https://github.com/elastic/kibana/commit/6f1b6907ba010b40a688f99def1c8fa497949008) |
| --- | --- | --- | --- |
| default | 45825 | +3 | 45822 |
| oss | 26565 | +3 | 26562 |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/8c209cdaa61749d17a34ab8f38ac5db5c0f1802b) | diff | [baseline](https://github.com/elastic/kibana/commit/6f1b6907ba010b40a688f99def1c8fa497949008) |
| --- | --- | --- | --- |
| core | 955.6KB | -131.4KB | 1.1MB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77691/","number":"77691","commit":"8c209cdaa61749d17a34ab8f38ac5db5c0f1802b"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78401/)
* Commit: 140e7dbb102eaa8bac04bc5e98a73a07d0817a4e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/140e7dbb102eaa8bac04bc5e98a73a07d0817a4e) | diff | [baseline](https://github.com/elastic/kibana/commit/5fd785a5b117285c2d3b97f2b8f3e087c0a4150a) |
| --- | --- | --- | --- |
| `core` | 659.4KB | +23.0B | 659.4KB |

### History
* :green_heart: [Build #77691](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77691/) succeeded 8c209cdaa61749d17a34ab8f38ac5db5c0f1802b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77691/","number":"77691","commit":"8c209cdaa61749d17a34ab8f38ac5db5c0f1802b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78401/","number":"78401","commit":"140e7dbb102eaa8bac04bc5e98a73a07d0817a4e"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-platform (Team:Platform)

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/)
* Commit: 08aeaf6cca585e05a05d367b629d565457d1433e
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/execution/node/626/log)
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/execution/node/776/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/testReport)

<details><summary>X-Pack EPM API Integration Tests.x-pack/test/ingest_manager_api_integration/apis/fleet/agents/complete_flow·ts.Ingest Manager Endpoints Fleet Endpoints fleet_agent_flow should work</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/testReport//junit/X-Pack%20EPM%20API%20Integration%20Tests/x-pack_test_ingest_manager_api_integration_apis_fleet_agents_complete_flow·ts/Ingest_Manager_Endpoints_Fleet_Endpoints_fleet_agent_flow_should_work)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 3 times on tracked branches: https://github.com/elastic/kibana/issues/75241

[00:00:00]       │
[00:00:00]         └-: Ingest Manager Endpoints
[00:00:00]           └-> "before all" hook
[00:00:09]           └-: Fleet Endpoints
[00:00:09]             └-> "before all" hook
[00:01:35]             └-: fleet_agent_flow
[00:01:35]               └-> "before all" hook
[00:01:35]               └-> "before all" hook
[00:01:35]                 │ info [empty_kibana] Loading "mappings.json"
[00:01:35]                 │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/I1VEYD3KTzK5R0sw5JvSjg] deleting index
[00:01:35]                 │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_1/wDh_0nHySDaeZPGTi70mdg] deleting index
[00:01:35]                 │ info [empty_kibana] Deleted existing index [".kibana_2",".kibana_1"]
[00:01:35]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana] creating index, cause [api], templates [], shards [1]/[1]
[00:01:35]                 │ info [empty_kibana] Created index ".kibana"
[00:01:35]                 │ debg [empty_kibana] ".kibana" settings {"index":{"number_of_replicas":"1","number_of_shards":"1"}}
[00:01:35]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana/RGrD0XSuTNSbcNxM-ruvMg] update_mapping [_doc]
[00:01:35]                 │ debg Migrating saved objects
[00:01:35]                 │ proc [kibana]   log   [22:48:02.484] [info][savedobjects-service] Creating index .kibana_2.
[00:01:35]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2] creating index, cause [api], templates [], shards [1]/[1]
[00:01:35]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updating number_of_replicas to [0] for indices [.kibana_2]
[00:01:35]                 │ proc [kibana]   log   [22:48:02.545] [info][savedobjects-service] Reindexing .kibana to .kibana_1
[00:01:35]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_1] creating index, cause [api], templates [], shards [1]/[1]
[00:01:35]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updating number_of_replicas to [0] for indices [.kibana_1]
[00:01:35]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.tasks] creating index, cause [auto(task api)], templates [], shards [1]/[1]
[00:01:35]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updating number_of_replicas to [0] for indices [.tasks]
[00:01:35]                 │ info [o.e.t.LoggingTaskListener] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] 6367 finished with response BulkByScrollResponse[took=2.8ms,timed_out=false,sliceId=null,updated=0,created=0,deleted=0,batches=0,versionConflicts=0,noops=0,retries=0,throttledUntil=0s,bulk_failures=[],search_failures=[]]
[00:01:35]                 │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana/RGrD0XSuTNSbcNxM-ruvMg] deleting index
[00:01:35]                 │ proc [kibana]   log   [22:48:02.910] [info][savedobjects-service] Migrating .kibana_1 saved objects to .kibana_2
[00:01:35]                 │ proc [kibana]   log   [22:48:02.915] [info][savedobjects-service] Pointing alias .kibana to .kibana_2.
[00:01:35]                 │ proc [kibana]   log   [22:48:02.939] [info][savedobjects-service] Finished in 456ms.
[00:01:35]               └-> "before all" hook
[00:01:35]                 │ info [docker:registry] 2020/10/02 22:48:02 source.ip: 172.17.0.1:44746, url.original: /search?package=endpoint&internal=true&experimental=true
[00:01:35]                 │ info [docker:registry] 2020/10/02 22:48:02 source.ip: 172.17.0.1:44748, url.original: /search?package=system&internal=true&experimental=true
[00:01:35]                 │ info [docker:registry] 2020/10/02 22:48:02 source.ip: 172.17.0.1:44754, url.original: /search?package=system&internal=true&experimental=true
[00:01:35]                 │ info [docker:registry] 2020/10/02 22:48:02 source.ip: 172.17.0.1:44756, url.original: /search?package=endpoint&internal=true&experimental=true
[00:01:35]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:35]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44760, url.original: /package/endpoint/0.13.1
[00:01:35]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44764, url.original: /package/endpoint/0.13.1/
[00:01:35]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:35]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44768, url.original: /package/system/0.5.3
[00:01:35]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44772, url.original: /package/system/0.5.3/
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44782, url.original: /search
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44780, url.original: /search
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44790, url.original: /package/endpoint/0.13.1
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44792, url.original: /package/system/0.5.3
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44796, url.original: /package/endpoint/0.13.1
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44798, url.original: /package/system/0.5.3
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44808, url.original: /package/system/0.5.3/
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44806, url.original: /package/endpoint/0.13.1/
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44812, url.original: /package/endpoint/0.13.1/
[00:01:36]                 │ info [docker:registry] 2020/10/02 22:48:03 source.ip: 172.17.0.1:44814, url.original: /package/system/0.5.3/
[00:01:36]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:37]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:37]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:37]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:41]                 │ info [docker:registry] 2020/10/02 22:48:08 source.ip: 172.17.0.1:44830, url.original: /search?package=system&internal=true&experimental=true
[00:01:41]                 │ info [docker:registry] 2020/10/02 22:48:08 source.ip: 172.17.0.1:44832, url.original: /package/system/0.5.3
[00:01:41]                 │ info [docker:registry] 2020/10/02 22:48:08 source.ip: 172.17.0.1:44836, url.original: /package/system/0.5.3/
[00:01:41]                 │ info [docker:registry] 2020/10/02 22:48:08 source.ip: 172.17.0.1:44844, url.original: /package/system/0.5.3
[00:01:41]                 │ info [docker:registry] 2020/10/02 22:48:08 source.ip: 172.17.0.1:44842, url.original: /search?package=system&internal=true&experimental=true
[00:01:41]                 │ info [docker:registry] 2020/10/02 22:48:08 source.ip: 172.17.0.1:44848, url.original: /package/system/0.5.3/
[00:01:41]                 │ info [docker:registry] 2020/10/02 22:48:08 source.ip: 172.17.0.1:44852, url.original: /package/system/0.5.3
[00:01:41]                 │ info [docker:registry] 2020/10/02 22:48:08 source.ip: 172.17.0.1:44856, url.original: /package/system/0.5.3/
[00:01:41]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:43]                 │ info [docker:registry] 2020/10/02 22:48:10 source.ip: 172.17.0.1:44868, url.original: /search?package=system&internal=true&experimental=true
[00:01:43]                 │ info [docker:registry] 2020/10/02 22:48:10 source.ip: 172.17.0.1:44870, url.original: /search?package=endpoint&internal=true&experimental=true
[00:01:43]                 │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updated role [fleet_enroll]
[00:01:43]                 │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updated user [fleet_enroll]
[00:01:45]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:45]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:46]               └-> should work
[00:01:46]                 └-> "before each" hook: global before each
[00:01:46]                 └-> "before each" hook: beforeSetupWithDockerRegistry
[00:01:46]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:49]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/3vSNgq2cSDmeOV-LAesVSw] update_mapping [_doc]
[00:01:55]                 └- ✖ fail: Ingest Manager Endpoints Fleet Endpoints fleet_agent_flow should work
[00:01:55]                 │      Error: socket hang up
[00:01:55]                 │       at createHangUpError (_http_client.js:332:15)
[00:01:55]                 │       at Socket.socketOnEnd (_http_client.js:435:23)
[00:01:55]                 │       at endReadableNT (_stream_readable.js:1145:12)
[00:01:55]                 │       at process._tickCallback (internal/process/next_tick.js:63:19)
[00:01:55]                 │ 
[00:01:55]                 │ 

```

#### Stack Trace
```
{ Error: socket hang up
    at createHangUpError (_http_client.js:332:15)
    at Socket.socketOnEnd (_http_client.js:435:23)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19) code: 'ECONNRESET', response: undefined }
```
</details>

---

<details><summary>X-Pack EPM API Integration Tests.x-pack/test/ingest_manager_api_integration/apis/fleet/agents/complete_flow·ts.Ingest Manager Endpoints Fleet Endpoints fleet_agent_flow should work</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/testReport//junit/X-Pack%20EPM%20API%20Integration%20Tests/x-pack_test_ingest_manager_api_integration_apis_fleet_agents_complete_flow·ts/Ingest_Manager_Endpoints_Fleet_Endpoints_fleet_agent_flow_should_work_2)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 3 times on tracked branches: https://github.com/elastic/kibana/issues/75241

[00:00:00]       │
[00:00:00]         └-: Ingest Manager Endpoints
[00:00:00]           └-> "before all" hook
[00:00:09]           └-: Fleet Endpoints
[00:00:09]             └-> "before all" hook
[00:01:32]             └-: fleet_agent_flow
[00:01:32]               └-> "before all" hook
[00:01:32]               └-> "before all" hook
[00:01:32]                 │ info [empty_kibana] Loading "mappings.json"
[00:01:32]                 │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/cQbsCOnIRSWGTah3dTqlFw] deleting index
[00:01:32]                 │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_1/ovHW8y9TS7eqKETx6FkKZw] deleting index
[00:01:32]                 │ info [empty_kibana] Deleted existing index [".kibana_2",".kibana_1"]
[00:01:32]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana] creating index, cause [api], templates [], shards [1]/[1]
[00:01:32]                 │ info [empty_kibana] Created index ".kibana"
[00:01:32]                 │ debg [empty_kibana] ".kibana" settings {"index":{"number_of_replicas":"1","number_of_shards":"1"}}
[00:01:32]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana/FA4hW-WPSQeKg-E1YVaVkA] update_mapping [_doc]
[00:01:32]                 │ debg Migrating saved objects
[00:01:32]                 │ proc [kibana]   log   [21:51:58.116] [info][savedobjects-service] Creating index .kibana_2.
[00:01:32]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2] creating index, cause [api], templates [], shards [1]/[1]
[00:01:32]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updating number_of_replicas to [0] for indices [.kibana_2]
[00:01:32]                 │ proc [kibana]   log   [21:51:58.190] [info][savedobjects-service] Reindexing .kibana to .kibana_1
[00:01:32]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_1] creating index, cause [api], templates [], shards [1]/[1]
[00:01:32]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updating number_of_replicas to [0] for indices [.kibana_1]
[00:01:32]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.tasks] creating index, cause [auto(task api)], templates [], shards [1]/[1]
[00:01:32]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updating number_of_replicas to [0] for indices [.tasks]
[00:01:32]                 │ info [o.e.t.LoggingTaskListener] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] 6327 finished with response BulkByScrollResponse[took=3.1ms,timed_out=false,sliceId=null,updated=0,created=0,deleted=0,batches=0,versionConflicts=0,noops=0,retries=0,throttledUntil=0s,bulk_failures=[],search_failures=[]]
[00:01:32]                 │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana/FA4hW-WPSQeKg-E1YVaVkA] deleting index
[00:01:32]                 │ proc [kibana]   log   [21:51:58.575] [info][savedobjects-service] Migrating .kibana_1 saved objects to .kibana_2
[00:01:32]                 │ proc [kibana]   log   [21:51:58.581] [info][savedobjects-service] Pointing alias .kibana to .kibana_2.
[00:01:32]                 │ proc [kibana]   log   [21:51:58.608] [info][savedobjects-service] Finished in 494ms.
[00:01:32]               └-> "before all" hook
[00:01:32]                 │ info [docker:registry] 2020/10/02 21:51:58 source.ip: 172.17.0.1:36678, url.original: /search?package=endpoint&internal=true&experimental=true
[00:01:32]                 │ info [docker:registry] 2020/10/02 21:51:58 source.ip: 172.17.0.1:36676, url.original: /search?package=system&internal=true&experimental=true
[00:01:32]                 │ info [docker:registry] 2020/10/02 21:51:58 source.ip: 172.17.0.1:36684, url.original: /search?package=system&internal=true&experimental=true
[00:01:32]                 │ info [docker:registry] 2020/10/02 21:51:58 source.ip: 172.17.0.1:36686, url.original: /search?package=endpoint&internal=true&experimental=true
[00:01:33]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:33]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:58 source.ip: 172.17.0.1:36690, url.original: /package/endpoint/0.13.1
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:58 source.ip: 172.17.0.1:36694, url.original: /package/endpoint/0.13.1/
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:58 source.ip: 172.17.0.1:36704, url.original: /package/system/0.5.3
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:58 source.ip: 172.17.0.1:36708, url.original: /package/system/0.5.3/
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36720, url.original: /search
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36718, url.original: /search
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36734, url.original: /package/system/0.5.3
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36736, url.original: /package/system/0.5.3
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36732, url.original: /package/endpoint/0.13.1
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36730, url.original: /package/endpoint/0.13.1
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36742, url.original: /package/system/0.5.3/
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36746, url.original: /package/system/0.5.3/
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36750, url.original: /package/endpoint/0.13.1/
[00:01:33]                 │ info [docker:registry] 2020/10/02 21:51:59 source.ip: 172.17.0.1:36752, url.original: /package/endpoint/0.13.1/
[00:01:33]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:34]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:34]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:34]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:38]                 │ info [docker:registry] 2020/10/02 21:52:04 source.ip: 172.17.0.1:36960, url.original: /search?package=system&internal=true&experimental=true
[00:01:38]                 │ info [docker:registry] 2020/10/02 21:52:04 source.ip: 172.17.0.1:36962, url.original: /package/system/0.5.3
[00:01:38]                 │ info [docker:registry] 2020/10/02 21:52:04 source.ip: 172.17.0.1:36966, url.original: /package/system/0.5.3/
[00:01:38]                 │ info [docker:registry] 2020/10/02 21:52:04 source.ip: 172.17.0.1:36972, url.original: /search?package=system&internal=true&experimental=true
[00:01:38]                 │ info [docker:registry] 2020/10/02 21:52:04 source.ip: 172.17.0.1:36974, url.original: /package/system/0.5.3
[00:01:38]                 │ info [docker:registry] 2020/10/02 21:52:04 source.ip: 172.17.0.1:36978, url.original: /package/system/0.5.3/
[00:01:38]                 │ info [docker:registry] 2020/10/02 21:52:04 source.ip: 172.17.0.1:36982, url.original: /package/system/0.5.3
[00:01:38]                 │ info [docker:registry] 2020/10/02 21:52:04 source.ip: 172.17.0.1:36986, url.original: /package/system/0.5.3/
[00:01:38]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:40]                 │ info [docker:registry] 2020/10/02 21:52:06 source.ip: 172.17.0.1:37070, url.original: /search?package=system&internal=true&experimental=true
[00:01:40]                 │ info [docker:registry] 2020/10/02 21:52:06 source.ip: 172.17.0.1:37068, url.original: /search?package=endpoint&internal=true&experimental=true
[00:01:40]                 │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updated role [fleet_enroll]
[00:01:40]                 │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] updated user [fleet_enroll]
[00:01:42]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:42]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:43]               └-> should work
[00:01:43]                 └-> "before each" hook: global before each
[00:01:43]                 └-> "before each" hook: beforeSetupWithDockerRegistry
[00:01:43]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:46]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-debian-tests-xxl-1601670360385125807] [.kibana_2/BfN3B6pvQmOiRP6f9eDTEw] update_mapping [_doc]
[00:01:52]                 └- ✖ fail: Ingest Manager Endpoints Fleet Endpoints fleet_agent_flow should work
[00:01:52]                 │      Error: socket hang up
[00:01:52]                 │       at createHangUpError (_http_client.js:332:15)
[00:01:52]                 │       at Socket.socketOnEnd (_http_client.js:435:23)
[00:01:52]                 │       at endReadableNT (_stream_readable.js:1145:12)
[00:01:52]                 │       at process._tickCallback (internal/process/next_tick.js:63:19)
[00:01:52]                 │ 
[00:01:52]                 │ 

```

#### Stack Trace
```
{ Error: socket hang up
    at createHangUpError (_http_client.js:332:15)
    at Socket.socketOnEnd (_http_client.js:435:23)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19) code: 'ECONNRESET', response: undefined }
```
</details>

---

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/08aeaf6cca585e05a05d367b629d565457d1433e) | [after](https://github.com/elastic/kibana/commit/00d2105650b2a3a22135cbfdcb4aee9310979b0b) | diff |
| --- | --- | --- | --- |
| `core` | 660.6KB | 660.6KB | +23.0B |

### History
* :green_heart: [Build #78401](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78401/) succeeded 140e7dbb102eaa8bac04bc5e98a73a07d0817a4e
* :green_heart: [Build #77691](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77691/) succeeded 8c209cdaa61749d17a34ab8f38ac5db5c0f1802b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77691/","number":"77691","commit":"8c209cdaa61749d17a34ab8f38ac5db5c0f1802b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78401/","number":"78401","commit":"140e7dbb102eaa8bac04bc5e98a73a07d0817a4e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/","number":"79325","commit":"08aeaf6cca585e05a05d367b629d565457d1433e"}]}
PIPELINE-->

vigneshshanmugam reviewed (approved):
LGTM with the mentioned comment.

restrry commented on packages/kbn-apm-config-loader/src/config.ts:
```diff
@@ -30,8 +30,13 @@ const getDefaultConfig = (isDistributable: boolean): ApmAgentConfig => {
       active: false,
       globalLabels: {},
+      centralConfig: false,
+      captureHeaders: false,
```
Could you add a comment explaining why it's disabled?

joshdover replied:
I am erring on the side of caution here and trying to avoid capturing any data that could be sensitive. We could be more selective about this and capture only headers we know to be safe (an allowlist of headers). WDYT?

joshdover replied:
Locked these down in production for now 👍

restrry commented on packages/kbn-apm-config-loader/src/config.ts:
```diff
@@ -30,8 +30,13 @@ const getDefaultConfig = (isDistributable: boolean): ApmAgentConfig => {
       globalLabels: {},
+      centralConfig: false,
+      captureHeaders: false,
+      captureBody: false,
```
1.in `elatic-apm-node/index.d.ts`:
```typescript
type CaptureBody = 'off' | 'errors' | 'transactions' | 'all';
```
2. is it possible to improve type safety of https://github.com/elastic/kibana/blob/aa9774faa08a2d85a45f5de35fbcbcee738dac5d/packages/kbn-apm-config-loader/src/types.ts#L24 ?

joshdover replied:
@pgayvallet I believe we were needing some type changes from APM agent for this correct?

restrry replied:
AgentConfigOptions is not exported from `@elastic/apm-rum`, but we could declare a compatible interface, maybe

joshdover replied:
Will handle in follow up. Should not affect 7.10 since we still need to add config-schema validation for 'productionizing' this feature.

restrry commented on src/apm.js:
```diff
@@ -36,7 +36,18 @@ module.exports = function (serviceName = name) {
+  const apm = require('elastic-apm-node');
+
+  // Filter out username PII
+  apm.addFilter((payload) => {
```
Could you add a comment that user data are populated automatically?
What kind of data are considered sensitive? Is email not classified as PII?

joshdover replied:
Anything that can trace back to a specific user is PII. Maybe it would be best to just change all `user` keys to `[REDACTED]`. I didn't test with other security realms that could contain other data besides username

joshdover replied:
I should add that things that are not considered PII in this context include IP address and mac address since they may be used by more than one person and are not necessarily static.

joshdover replied:
>  it would be best to just change all user keys to [REDACTED]

done.

restrry commented on src/core/public/apm_system.ts:
```diff
@@ -39,10 +40,10 @@ export class ApmSystem {
    */
   constructor(private readonly apmConfig?: ApmConfig) {
-    this.enabled = process.env.IS_KIBANA_DISTRIBUTABLE !== 'true' && apmConfig != null;
+    this.enabled = apmConfig != null && !!apmConfig.active;
```
I had a hard time debugging the config. It's turned out we don't read values from `kibana.dev.yml`. It might be another source of confusion for developers.

joshdover replied:
> It's turned out we don't read values from `kibana.dev.yml`

I think there was some confusion on whether or not we still supported this. It appears we do. Can fix as a follow up so I can get this in for FF

joshdover replied:
Opened issue: https://github.com/elastic/kibana/issues/79490
Should be a pretty quick follow up

restrry commented on packages/kbn-apm-config-loader/src/config.ts:
```diff
@@ -30,8 +30,13 @@ const getDefaultConfig = (isDistributable: boolean): ApmAgentConfig => {
+      centralConfig: false,
+      captureHeaders: false,
+      captureBody: false,
+      logUncaughtExceptions: true,
```
Note: it's possible to set 
```
elastic.apm.captureBody: 'all'
elastic.apm.captureHeaders: true
```
in config. Shouldn't we forbid this to ensure PII isn't exposed?

joshdover replied:
I wasn't sure how strict we should be here. Though this isn't a supported customer-facing feature yet, we may document it in the future. If so, I thought keeping this as flexible as possible for customers would be best. But we're not building this for that use case right now, so maybe it would be better to just lock it down?

joshdover replied:
Locked this down

joshdover commented:
Pushed updates that addressed comments here and also fixed the APM serviceName in development (was broken by https://github.com/elastic/kibana/pull/79379, resulting in the main server and basePath proxy server both reporting as the "kibana-proxy" service)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79699/)
* Commit: fa9345333a2d6f5cb50db9ffe1ac53375f9ff857

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/fa9345333a2d6f5cb50db9ffe1ac53375f9ff857) | [after](https://github.com/elastic/kibana/commit/59e4e0631608ae58d933864a5b968511a13f2ec9) | diff |
| --- | --- | --- | --- |
| `core` | 661.2KB | 661.2KB | +23.0B |

### History
* :broken_heart: [Build #79325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/) failed 08aeaf6cca585e05a05d367b629d565457d1433e
* :green_heart: [Build #78401](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78401/) succeeded 140e7dbb102eaa8bac04bc5e98a73a07d0817a4e
* :green_heart: [Build #77691](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77691/) succeeded 8c209cdaa61749d17a34ab8f38ac5db5c0f1802b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77691/","number":"77691","commit":"8c209cdaa61749d17a34ab8f38ac5db5c0f1802b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78401/","number":"78401","commit":"140e7dbb102eaa8bac04bc5e98a73a07d0817a4e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79325/","number":"79325","commit":"08aeaf6cca585e05a05d367b629d565457d1433e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79699/","number":"79699","commit":"fa9345333a2d6f5cb50db9ffe1ac53375f9ff857"}]}
PIPELINE-->
